### PR TITLE
[WIP]urlPrefix should be unique

### LINKED ISSUE #116: [Bug] urlPrefix should be unique (add index)

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #116

### CODE REVIEW

File: lib/abstractions/create.js [hunk 1]
```diff
@@ -48,8 +48,32 @@ module.exports = options => {
...
-    var savedRecord = await pendingRecord.save(),
-        body = {};
+    try {
+      var savedRecord = await pendingRecord.save();
+    } catch ( err ) {
+      let error = err.message;
+
+      if (err.code === 11000){
+        let path = /index:\s([^\d_]*).*"(\w+)/i.exec(error)[1];
+        let val = /index:\s([^\d_]*).*"(\w+)/i.exec(error)[2];
```
Comment (James1x0) [inline lib/abstractions/create.js]: Match this once, then use the matches. Here you are needlessly instantiating and running two regex statements. Also combine variable dec

File: lib/abstractions/create.js [hunk 2]
```diff
@@ -48,8 +48,32 @@ module.exports = options => {
...
+        let errors = [];
+
+        errors.push({
+          status: 400,
+          title: 'duplicate Key Error',
+          detail: 'duplicate URL Prefix',
```
Comment (James1x0) [inline lib/abstractions/create.js]: Please capitalize `duplicate` in title and detail.

File: lib/abstractions/create.js [hunk 3]
```diff
@@ -48,8 +48,32 @@ module.exports = options => {
...
+          path,
+          val
+        });
+
+        ctx.status = 400;
+        ctx.body = { errors };
+        return;
+      }
+      ctx.status = 500;
```
Comment (James1x0) [inline lib/abstractions/create.js]: You need to throw the error here.

File: lib/abstractions/create.js [hunk 4]
```diff
@@ -48,8 +48,32 @@ module.exports = options => {
...
+    }
+
+    let   body = {};
```
Comment (James1x0) [inline lib/abstractions/create.js]: 👎

File: lib/abstractions/create.js [hunk 5]
```diff
@@ -48,8 +48,32 @@ module.exports = options => {
       return;
     }
 
-    var savedRecord = await pendingRecord.save(),
-        body = {};
+    try {
+      var savedRecord = await pendingRecord.save();
+    } catch ( err ) {
+      let error = err.message;
```
Comment (James1x0) [inline lib/abstractions/create.js]: Are you using this?

File: lib/abstractions/create.js (hunk 3 above)
Comment (JavaLavaMT) [reply lib/abstractions/create.js]: like this?     
 ```
ctx.status = 500;
 throw err;

```

**Pull request merged**